src/Databases/DatabaseOnDisk.cpp

```diff
@@ -387,17 +389,27 @@ void DatabaseOnDisk::checkMetadataFilenameAvailability(const String & to_table_n
 
 void DatabaseOnDisk::checkMetadataFilenameAvailabilityUnlocked(const String & to_table_name) const
 {
+    // Compute allowed max length directly
+    size_t allowed_max_length = computeMaxTableNameLength(database_name, getContext());
     String table_metadata_path = getObjectMetadataPath(to_table_name);
 
+    if (escapeForFileName(to_table_name).length() > allowed_max_length)
+        throw Exception(ErrorCodes::ARGUMENT_OUT_OF_BOUND,
+                        "The max length of table name for database {} is {}, current length is {}",
```

Comment:
This exception message may be confusing in case of table names that need escaping because it makes the filename much longer, `escapeForFileName(to_table_name).length() != to_table_name.length()` in this case